internal/controllers/machinedeployment/machinedeployment_rollout_planner_test.go

```diff
@@ -620,27 +621,21 @@ func machineSetControllerMutator(log *fileLogger, ms *clusterv1.MachineSet, scop
 					return errors.Errorf("[MS controller] - PANIC! %s is set to send replicas to %s, which does not exists", ms.Name, targetMSName)
 				}
 
-				// Limit the number of machines to be moved to avoid to exceed the final number of replicas for the target MS.
-				// TODO(in-place): consider using the DesiredReplicasAnnotation which is propagated together with the other decisions of the rollout planner
-				//  another option to consider is to shift this check on the newMS (so we perform move, but we don't start the in-place upgrade).
-				machinesToMove := min(machinesToDeleteOrMove, ptr.Deref(scope.machineDeployment.Spec.Replicas, 0)-ptr.Deref(targetMS.Spec.Replicas, 0))
```

Comment:
As discussed during review, this would have been racy when implemented in the prod code for the oldMS reconciliation.

So now we are moving all the machines, and deferring to the newMS to get rid of exceeding machines.
This leads to the ideal results - perform the minimal number of in place upgrades/avoid unnecessary in place updates - under the assumption that machines upgrading in place have the highest deletion priority.

Notes about how to change the prod code have been changed accordingly